internal/services/appservice/helpers/windows_web_app_schema.go

```diff
@@ -644,14 +644,25 @@ func (s *SiteConfigWindows) ExpandForUpdate(metadata sdk.ResourceMetaData, exist
 		expanded.AppCommandLine = pointer.To(s.AppCommandLine)
 	}
 
+	if metadata.ResourceData.HasChange("site_config.0.health_check_eviction_time_in_min") {
+		if appSettings == nil {
+			appSettings = make(map[string]string)
+		}
+		appSettings["WEBSITE_HEALTHCHECK_MAXPINGFAILURES"] = strconv.Itoa(int(s.HealthCheckEvictionTime))
+	}
+
 	if len(s.ApplicationStack) == 1 {
 		winAppStack := s.ApplicationStack[0]
-		if metadata.ResourceData.HasChange("site_config.0.application_stack.0.node_version") {
+
+		// (@apokalypt) If we are using node we need to always set the version since we remove it when we are parsing the web app
+		// https://github.com/hashicorp/terraform-provider-azurerm/issues/25452
+		if winAppStack.NodeVersion != "" {
```

Comment:
We should still be using `d.HasChanges` here to support the use of `ignore_changes`. Can we please amend this to.
```suggestion

		if metadata.ResourceData.HasChange("site_config.0.application_stack.0.node_version") || winAppStack.NodeVersion != "" {
```

Could we also remove the comment, since that information should be traceable in the git history, PR description and issue links. To ensure that this peculiarity is maintained we should add a test for this.